enhancements/evidence-forwarding.md

```diff
@@ -0,0 +1,305 @@
+also be deleted like any other cluster resource.
+
+A improvement to the idea of putting evidence reports on disk would be to dump
+them into an object or file store. This has an added benefit of allowing
+administrators managing multiple clusters to put evidence in a single location.
+
+## Motivation
+
+The motivation of the enhancement is to store compliance evidence outside the
+cluster and its storage pools.
+
+### Goals
+
+The goal is to provide a way for Kubernetes administrators to configure the
+Compliance Operator to send compliance reports, specifically ARF reports, to an
+external storage pool, like an object or file store.
+
+### Non-Goals
+
+It is not the goal of this enhancement to implement results forwarding, or
+adjust the definition of evidence outside an ARF report.
+
+## Proposal
+
+Provide an additional configuration for scan settings that allows site
+reliability engineers to forward evidence to a storage pool. Today, the
+Compliance Operator provides a `default` `ScanSetting` that integrates with
+persistent storage:
+
+```yaml
+---
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSetting
+metadata:
+  name: default
+  namespace: openshift-compliance
+rawResultStorage:
+  pvAccessModes:
+  - ReadWriteOnce
+  rotation: 3
+  size: 1Gi
+roles:
+- worker
+- master
+scanTolerations:
+  default:
+  - operator: Exists
+  schedule: 0 1 * * *
+```
+
+A site reliability engineer could create another `ScanSetting` that bypasses
+storing evidence on physical volumes with the following:
+
+```yaml
+---
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSetting
+metadata:
+  name: external-storage
+  namespace: openshift-compliance
+externalResultStorage:
+  endpoint: https://object-storage.example.com/
+  tls:
+roles:
+- worker
+- master
+scanTolerations:
+  default:
+  - operator: Exists
+  schedule: 0 1 * * *
+```
+
+A site reliability engineer could configure both storage mechanisms for redundancy:
+
+```yaml
+---
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSetting
+metadata:
+  name: external-storage-and-cluster-storage
+  namespace: openshift-compliance
+rawResultStorage:
+  pvAccessModes:
+  - ReadWriteOnce
+  rotation: 3
+  size: 1Gi
+externalResultStorage:
+  endpoint: https://object-storage.example.com/
+  tls:
+roles:
+- worker
+- master
+scanTolerations:
+  default:
+  - operator: Exists
+  schedule: 0 1 * * *
+```
+
+### User Stories
+
+1. As a site reliability engineer, I want to configure the Compliance Operator
+   to send all ARF reports to a dedicated object store.
+2. As a site reliability engineer, I want to opt out of storing any ARF reports
+   on physical volumes.
+3. As a site reliability engineer, I expect verbose alerts when the Compliance
+   Operator cannot send ARF reports to a configured external storage pool.
+4. As a site reliability engineer, I expect the Compliance Operator to fail if
+   the configured endpoint is unreachable.
```

Comment:
Do we always want this failure mode?

It also seems not-Kube to have a payload for a remote service ready, and fail when it's unreachable (at least without retrying).

Reply:
Yeah - that's a good point. What if we break this apart into two separate failure modes that are more specific?

1.) As a site reliability engineer, I expect `SettingBinding` create and update operations to fail if the endpoint malformed
2.) As a site reliability engineer, I expect the compliance operator to retry persisting evidence to external storage based on configuration. If retrying fails, the persistence steps will fail and issue an alert.

Reply:
IMO both should retry, maybe the difference would be whether the failure to forward errors the scans? Unsure if the number of retry steps should be configurable (perhaps not initially)